docs/interfaces.md

```diff
@@ -0,0 +1,571 @@
+```
+
+```c++
+// Provided/implemented by a common logging module
+// interface exposed to TLCC_enclave via cgo
+void ocall_print_string([in, string] const char *str);
+```
+
+## State:
+TLCC keeps a local reference of the enclave id. This reference is volatile. Also a reference to the peer is kept to access the ledger and read blocks.
+
+```go
+type TrustedLedgerCC struct {
+	enclave enclave.Stub
+	peer    *peer.Peer
+}
+
+type StubImpl struct {
+	eid C.enclave_id_t
+}
+```
+
+```protobuf
+syntax = "proto3"
+
+// Setup and Close messages are used to establish a secure channel between
+// ECC_Enclave and TLCC_Enclave. Once a secure channel is established,
+// Session Requests to TLCC can be issued to invoke tlcc functions such
+// as getMetadata or verifyMSPId.
+
+// TODO How to integrate the the message authenticator?!
+// should we abstract it as signature here even though we may
+// use cmac once a session is established?
+
+message SessionSetup {
+        // TBD input enclave vk?
+        // TLCC Mrenclave?
+        // channel
+}
+
+message SessionSetupResponse {
+        SessionSetup setup = 1;
+        byte context       = 2;         // context that references the session
+}
+
+message SessionClose {
+        byte context = 1;
+}
+
+message SessionCloseResponse {
+}
+
+message SessionRequest {
+        byte context = 1;
+        byte nonce   = 2;
+        oneof request {
+                GetMetadataRequest metadata     = 1; 
+                validateMSPRequest validate_msp = 2;
+        }
+}
+
+message SessionRequeatResponse {
+        SessionRequest request = 1;
+        oneof response {
+                GetMetadataResponse metadata     = 1; 
+                validateMSPResponse validate_msp = 2; 
+        }
+}
+
+```
+
+
+# TLCC_Enclave
+
+## Interface:
+Defined in the TLCC EDL.
+```c++
+// TBD is this interface already sufficient to support multi-channel?
+// TODO check if a block contatains any channel information?
+
+// creates a new channel
+public int ecall_join_channel(
+        [in, size=gen_len] uint8_t *genesis, uint32_t gen_len);
+
+// pushes next block to tlcc
+public int ecall_next_block(
+        [user_check] uint8_t *block_bytes, uint32_t block_size);
+```
+
+```c++
+// this secure session functions are defined in a secure channel module
+//included from secure_channel.edl
+
+public int session_setup(
+        [in] setup SessionSetup,
+        [out] SessionSetupResponse);
+
+public int session_close(
+        [in] close SessionClose
+        [out] SessionCloseReponse);
+
+public int session_request(
+        [in] request SessionRequest,
+        [out] SessionRequestResponse;
+```
+
+```c++
+// addtionally we may use the state sealing module to persists TLCC state
+// see below the data sealing module.
+include sealed_storage.edl
+```
+
+Session requests are depatched to the following interfaces:
+```c++
+// TODO refine
+// verify state data
+public int get_state_metadata(
```

Comment:
Depending on how we handle state updates and consistency of requests inside a single transaction, we might have to surface some additional param here?

Reply:
Probably worth also mentioning that this namespace-enhanced api makes obsolete the separate `check_ercc_commited_data` function we had in the `fpc-key-dist` UML diagram.

Reply:
ok, will add TODO in the API document to show where UMLs need to be upgraded.